pkg/segment/aggregations/generateevents.go

```diff
@@ -0,0 +1,91 @@
+package aggregations
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+)
+
+func getFormattedTime(t time.Time) string {
+	return t.Format("Mon Jan 2 15:04:05 2006 -0700")
+}
+
+func createGenTimeEvent(start time.Time, end time.Time) map[string]interface{} {
+	return map[string]interface{}{
+		"starttime":  uint64(start.UnixMilli()) / 1000,
+		"endtime":    uint64(end.UnixMilli()) / 1000,
+		"starthuman": getFormattedTime(start),
+		"endhuman":   getFormattedTime(end),
+	}
+}
+
+func SetGeneratedCols(genEvent *structs.GenerateEvent, genCols []string) {
+	for _, col := range genCols {
+		genEvent.GeneratedCols[col] = true
+	}
+}
+
+func PerformGenTimes(aggs *structs.QueryAggregators) error {
+	if aggs.GenerateEvent.GenTimes == nil {
+		return fmt.Errorf("PerformGenTimes: GenTimes is nil")
+	}
+	if aggs.GenerateEvent.GenTimes.Interval == nil {
+		aggs.GenerateEvent.GenTimes.Interval = &structs.SpanLength{
+			Num:       1,
+			TimeScalr: utils.TMDay,
+		}
+	}
+	genCols := []string{"starttime", "endtime", "starthuman", "endhuman"}
+
+	SetGeneratedCols(aggs.GenerateEvent, genCols)
+
+	start := aggs.GenerateEvent.GenTimes.StartTime
+	end := aggs.GenerateEvent.GenTimes.EndTime
+	interval := aggs.GenerateEvent.GenTimes.Interval.Num
+	if interval < 0 {
+		start, end = end, start
+		interval = -interval
+	}
+
+	// No need to generate events
+	if start >= end {
+		return nil
+	}
```

Comment:
Shouldn't this be thrown as an error? What does Splunk do in this scenario?

Reply:
It just returns 0 results.